### ISSUE TITLE
Create a (TexInfo) user manual

### ISSUE THREAD

**Issue by bbatsov** [clojure-emacs/cider#689]
I'd like us to create a basic user manual for CIDER, that would be similar in spirit and scope to SLIME's [user manual](https://github.com/slime/slime/blob/master/doc/slime.texi). Ideally it'd be a texinfo document (as those are easy to read/search in Emacs), but I'm open to other alternatives (like generating the texinfo from another more popular format or publishing a web-only user manual on a dedicated CIDER website).

The document should be more or less an extended version of the README, documenting in a bit more details the important functionality available to the users (ideally it might describe a bit more). Once we have an initial version ready we'll always keep it in sync with the latest development.

Unfortunately I won't have time to work on this any time soon, so I'd to ask for your assistance to complete this task. I think the user manual is one of the biggest missing features at this point.

**Comment by vermiculus** [comment]
Let me know if/when/how I can help :smile:

**Comment by Malabarba** [comment]
I think one easy way to start is to convert the readme to a proper manual. 
Right now it's overflowing with information and a single markdown file is really not the best way to display so much content. 

As @bbatsov says, it's really a manual disguised as a huge readme. Except it's gained a few more pounds and that disguise fits about as well as an overweight 30 year-old man fits in a sailor moon costume.

 The good thing is that there's little need to add any content. Simply converting it to a proper manual would be a huge benefit. Then we could reduce the readme file to the bare minimum: installation,  some screenshots, and 10-20 links to the manual.

**Comment by Malabarba** [comment]
So the only question is what system to use for the manual. Probably, the person who takes up this task deserves an opinion on the matter. :-)

**Comment by vermiculus** [comment]
I'll get a version going in straight Texinfo. The Org exporter is…a bit less than advertised :smile: 

Unless there are objections/suggestions, of course.

**Comment by Malabarba** [comment]
I'm fine with texinfo, but we should hear @bbatsov too. 

It's important that the manual be available online because I think cider gets a lot of users who are new to Emacs, and the info browser wouldn't be as useful to them as a Web page. 

So if texinfo can be exported to html then that sounds fine to me. It should be trivial to host that in cider's gh-pages branch and recompile that before every release.

**Comment by vermiculus** [comment]
More information: http://www.gnu.org/software/texinfo/manual/texinfo/texinfo.html#Generating-HTML

**Comment by bbatsov** [comment]
I'm not sure that's a good idea. AFAIK you can't generate good looking markup from texinfo and in the age of the browsers it seems we'll be doing a mistake. Probably it makes more sense to pick a more "neutral" format which can be converted to HTML and texinfo. @lunaryorn did a lot of research on the subject a while back and I think we decided to go with http://sphinx-doc.org/

I'm open to alternatives.

**Comment by lunaryorn** [comment]
@bbatsov I'm back to Texinfo for Flycheck.  As good as Sphinx is, it's Texinfo support is definitely not as good as native Texinfo, and the Info manuals it produces frequently look broken.  And in the end, it wasn't worth the complexity, and the amount of baby-sitting it required (e.g. manually rebuilding the Texinfo file for MELPA to pick up for packaging, working around various glitches in Texinfo generation, etc.).

As bad as Texinfo is, it's the standard for Emacs, and everything else will always be inferior, unless we put a very significant effort into it.  And since none won't, we're going to be stuck with Texinfo (for good Info manuals) for a foreseeable time.

Texinfo's HTML output is far from perfect, with some CSS and custom post-processing it's [quite ok](http://www.flycheck.org/manual/0.23/Syntax-checkers.html#Syntax-checkers).  And if that's not sufficient, it's far easier to take the XML or Docbook output of Texinfo and feed it into a custom XSL stylesheet before trying to write a good Info generator for another format.

There are few source formats that can replace Texinfo anyway.  ReST is one, AsciiDoc probably another, but Markdown definitely falls short, because it's syntax is too primitive to encode the cross-referencing features of Texinfo.  You'd effectively have to write your own custom Markdown dialect, enriched by some Texinfo features, and that's going to be a lot of work.

TL;DR:  Use Texinfo.  It's the only thing that works for Emacs in my experience.

**Comment by Malabarba** [comment]
> Texinfo's HTML output is far from perfect, with some CSS and custom post-processing it's quite ok. 

How much custom post processing is involved in that?

I was thinking we might be able make it look at least as nice as the readme by just copying whatever stylesheet github uses (with small modifications). It would be simple, but comfortable to most people.

**Comment by lunaryorn** [comment]
@Malabarba Not much, take a look at the `Rakefile` of flycheck.github.io

**Comment by bbatsov** [comment]
> TL;DR: Use Texinfo. Whether you like it or not, it's effectively the only thing that works for Emacs in my experience.

I trust @lunaryorn's judgement. Let's go with Texinfo.

**Comment by vermiculus** [comment]
I'll continue work on it tonight and hopefully have a PR in by tomorrow.

**Comment by vermiculus** [comment]
Better late than never? :stuck_out_tongue:

**Comment by bbatsov** [comment]
Totally!

On 8 August 2015 at 06:15, Sean Allred <notifications@github.com> wrote:

> Better late than never? [image: :stuck_out_tongue:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/clojure-emacs/cider/issues/689#issuecomment-128893410>
> .
>



-- 
Best Regards,
Bozhidar Batsov

http://www.batsov.com

**Comment by bbatsov** [comment]
One more idea about the manual's format - https://github.com/GitbookIO/gitbook

I kind of like this, as we'll be able to generate both a website and an epub book.

**Comment by vermiculus** [comment]
Ah I remember that project… it worked pretty well a few years ago. You'd likely lose the Texinfo functionality, though.

**Comment by bbatsov** [comment]
True. It seems to me, however, that our top focus would be a good web manual and that other formats should be considered as a plus, not a must. Ideally, we'd have a Texinfo manual, but I'm not sure how much sense does it make to invest a lot of efforts into it.

**Comment by bbatsov** [comment]
So, the new manual is here - http://cider.readthedocs.org/en/latest/

I really need some help to polish and extend it. Down the road we should also figure out how to generate a pdf/epub/texinfo version from it, but at least we're moving in the right direction.
